src/jit/treelifeupdater.cpp

```diff
@@ -168,6 +183,13 @@ void TreeLifeUpdater<ForCodeGen>::UpdateLifeVar(GenTree* tree)
                             VarSetOps::AddElemD(compiler, stackVarDeltaSet, fldVarIndex);
                         }
                     }
+
+                    if (ForCodeGen)
+                    {
+                        // BRIAN:: see how scopes are reported for promoted structs. Are they considered in the
+                        // amount of local variables to report?
```

Comment:
This could be quite tricky, as I don't think there's a way to express, in the scope info, the notion that different fields of a struct live in different locations, e.g. if we had an instance s of a struct S that had 3 fields a, b, and c, we might have s.a in one register, s.b on the stack, and s.c in another register. I *think* that this would require modifying the format.

Reply:
How is it working now for debugging? It's all in the stack?

Reply:
Right - we currently don't promote fields of structs in debug mode, just as we don't enregister any local variables.

Reply:
Can we track this with an issue and a TODO comment?

Reply:
Sure we can!